test/EnsembleKalmanProcess/runtests.jl

```diff
@@ -79,6 +79,123 @@ inv_problems = [inv_problems..., nl_inv_problems...]
+    ekiobj = EKP.EnsembleKalmanProcess(
+        initial_ensemble,
+        y_obs,
+        Γy,
+        Inversion(),
+        scheduler = DefaultScheduler(),
+        accelerator = NesterovAccelerator(),
+    )
+    eksobj = EKP.EnsembleKalmanProcess(
+        initial_ensemble,
+        y_obs,
+        Γy,
+        Sampler(prior),
+        scheduler = DefaultScheduler(),
+        accelerator = NesterovAccelerator(),
+    )
+    ekiobj_noacc = EKP.EnsembleKalmanProcess(initial_ensemble, y_obs, Γy, Inversion(), scheduler = DefaultScheduler())
+    eksobj_noacc =
+        EKP.EnsembleKalmanProcess(initial_ensemble, y_obs, Γy, Sampler(prior), scheduler = DefaultScheduler())
+    ekiobj_noacc_specified = EKP.EnsembleKalmanProcess(
+        initial_ensemble,
+        y_obs,
+        Γy,
+        Inversion(),
+        scheduler = DefaultScheduler(),
+        accelerator = DefaultAccelerator(),
+    )
+    eksobj_noacc_specified = EKP.EnsembleKalmanProcess(
+        initial_ensemble,
+        y_obs,
+        Γy,
+        Sampler(prior),
+        scheduler = DefaultScheduler(),
+        accelerator = DefaultAccelerator(),
+    )
+
+    ## test EKP object's accelerator type is consistent (EKP constructor reassigns object in some cases)
+    @test typeof(ekiobj.accelerator) <: NesterovAccelerator
+    @test typeof(eksobj.accelerator) <: NesterovAccelerator
+    @test typeof(ekiobj_noacc.accelerator) <: DefaultAccelerator
+    @test typeof(eksobj_noacc.accelerator) <: DefaultAccelerator
+    @test typeof(ekiobj_noacc_specified.accelerator) <: DefaultAccelerator
+    @test typeof(eksobj_noacc_specified.accelerator) <: DefaultAccelerator
+
+    ## test NesterovAccelerators satisfy desired ICs
+    @test ekiobj.accelerator.r == 3.0
+    @test ekiobj.accelerator.u_prev == initial_ensemble
+    @test eksobj.accelerator.r == 3.0
+    @test eksobj.accelerator.u_prev == initial_ensemble
+
+    # test method convergence
+    # Note: this test only requires that the final ensemble is an improvement on the initial ensemble,
+    # NOT that the accelerated processes are more effective than the default, as this is not guaranteed.
+    # Specific cost values are printed to give an idea of acceleration.
+    processes = [Inversion(), Sampler(prior), TransformInversion(inv(Γy))]
+    T_end = 1 # (this could fail a test if N_iters is not enough to reach T_end)
+    for process in processes
+        accelerators = [DefaultAccelerator(), NesterovAccelerator()]
+        N_iters = [30, 30, 30]
+        init_means = []
+        final_means = []
+
+        for (accelerator, N_iter) in zip(accelerators, N_iters)
+            println("Accelerator: ", nameof(typeof(accelerator)), " Process: ", nameof(typeof(process)))
+            if !(nameof(typeof(process)) == Symbol(Unscented))
+                ekpobj = EKP.EnsembleKalmanProcess(
+                    initial_ensemble,
+                    y_obs,
+                    Γy,
+                    process,
```

Comment:
To be safe, I would create a new process object for each test. There is no guarantee that they aren't modified during the calls to EKP updates.